support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -376,7 +376,17 @@ function CheckoutComponent({ geoId }: Props) {
 	const showStateSelect =
 		countryId === 'US' || countryId === 'CA' || countryId === 'AU';
 
+	/**
+	 * Payment method: PayPal
+	 * BAID = Billing Agreement ID
+	 */
 	const [paymentMethod, setPaymentMethod] = useState<PaymentMethod>();
+	const [payPalLoaded, setPayPalLoaded] = useState(false);
+	useEffect(() => {
+		if (paymentMethod === 'PayPal' && !payPalLoaded) {
+			void loadPayPalRecurring().then(() => setPayPalLoaded(true));
+		}
+	}, [paymentMethod, payPalLoaded]);
```

Comment:
Is there a reason we've separated this from `payPalBAID`?

Reply:
Just seemed  neater to move useEffect's underneath the associated/related useStates (ie batched together).

Reply:
Rather than neatness I tend to group things by functionality e.g. PayPal, Stripe etc. 

The reason being is that I find myself rarely working on all side-effects, but rather updating Stripe, or similar, so having the code co-located on that basis often avoids having to bounce around the code making it a little unwieldy.

Maybe we should look at starting to split this file up?

Reply:
Let me revert this re-order firstly by functionality now.

Reply:
reverted